### ISSUE TITLE
[Bug Report][3.0.0-beta.13] Cannot create themes with custom color names and variations

### ISSUE THREAD

**Issue by c-harding** [vuetifyjs/vuetify#15850]
### Environment
**Vuetify Version:** 3.0.0-beta.13
**Vue Version:** 3.2.40
**Browsers:** Chrome 106.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
Define a theme, including non-standard color names. E.g. `theme.colors.color2 = '#abcdef'`.

In the call to createVuetify, pass a variation definition including one of these new colors. E.g. `options.theme.variations = { colors: ['color2'], lighten: 4, darken: 0 }`.

Start the development server, and open it in a web browser.

### Expected Behavior
the app should start, and the defined variations should be available.

### Actual Behavior
The page does not load in the browser, instead a fullscreen type error is displayed: Colors can only be numbers or strings, recieved undefined instead.

This is because the variations are calculated on the light and dark themes as well as the provided theme, where these colors are not available.

### Reproduction Link
<a href="https://codesandbox.io/s/recursing-spence-3kxwnr?file=/src/vuetify.js" target="_blank">https://codesandbox.io/s/recursing-spence-3kxwnr?file=/src/vuetify.js</a>

### Other comments
It should be possible to prevent the default themes from being provided

**Comment by KaelWD** [comment]
Solution: `themes: { light: theme, dark: theme },`
There's no point adding a custom theme if you aren't using the default ones

**Comment by c-harding** [comment]
Thanks @KaelWD. This should be documented: the docs should suggest modifying the existing themes rather than just defining (and setting as default) new themes.

https://next.vuetifyjs.com/en/features/theme/

**Comment by Dinuz** [comment]
@KaelWD I don't understand your answer at all. The idea of using a custom theme is exactly because you don't want to use a default one. What is the point of a custom theme otherwise? 
Let's say I have a specific brand palette, and I want to generate a theme using this palette, and this specific palette has a primary, a secondary and an accent color. So I cannot use the "variations" tool to automatically generate my lighten and darken variations for every custom color name (in my case accent), but I can use it to generate these variations for the colors that I named primary and secondary (exactly as you do in the code)?

Would be probably super cool if instead we could use also custom names inside the tool

**Comment by KaelWD** [comment]
I don't understand the question. If you name your custom theme `light` it'll override the default light theme colours.

**Comment by Dinuz** [comment]
@KaelWD I get that:) But naming my custom theme light will not allow to generate automatic variations anyway. The automatic variations generator (lighten, darken) works only with pre-existing color names (e.g primary, secondary) but not with custom color names (e.g accent, mistakes etc). In other words (and this I believe is the issue that was first reported by @c-harding ), if I follow the directions in the documentation, in side of theme, I can also have an object called variations, and this object works only with predefined color names.

So This will not work:

`export default createVuetify(
    {
        theme: {
            defaultTheme: 'myTheme',
            themes: {
                light: myTheme
            },
            variations: {
                colors: ['primary', 'secondary', 'accent'],
                lighten: 5,
                darken: 4,
            },
        },
    }
)`

But this will work instead:
`export default createVuetify(
    {
        theme: {
            defaultTheme: 'myTheme',
            themes: {
                light: myTheme
            },
            variations: {
                colors: ['primary', 'secondary'],
                lighten: 5,
                darken: 4,
            },
        },
    }
)`

So the only way that I have to add my own custom color + variations is to explicitly add them in my theme (but this is time consuming and repetitive).

Could we modify the code behind "variations" to allow also color names that are custom?
This would be really a useful feature, cause you will just need to input the base colors in your palette, and have all the variations automatically generated. We could make it also a bit more powerful, and generates not only the darken and lighten variations, but also the so called "accents" as you have in the default material palette.

something that allow me to do this: 

`export default createVuetify(
    {
        theme: {
            defaultTheme: 'myTheme',
            themes: {
                light: myTheme
            },
            variations: {
                colors: ['primary', 'secondary', 'custom-name'],
                lighten: 5,
                darken: 4,
                accent:3,
            },
        },
    }
)`

and get out custom-name-darken-1, custom-name-darken-2, custom-name-lighten-1, custom-name-accent-1 etc.


Is this now more clear?

**Comment by c-harding** [comment]
@Dinuz try it again with `themes: { light: myTheme, dark: myTheme }`. Both need to be specified for it to work. I do agree that this behaviour is unexpected though: it would be good to be able to turn off dark mode support completely

**Comment by Dinuz** [comment]
@c-harding OMG you are right, it work indeed if you explicitly define both. I agree with you, It is a bit weird the behavior, and certainly can cause confusion:) Probably explicitly documenting it, it's the best solution as you suggested. 

@KaelWD so the issue is solved now that @c-harding explicitly clarified the solution, and I thank both for that.
However the suggestion of creating a function to generate also the 'accents' for a color still remain valid.

So do we know what is the mathematical relationship (as we do for lighten and darken) between a base color and it's accent shade (A100, A200, A400 and A700)? If we do (and doesn't involve manual tweak to calibrate the tones) we could add the 3rd option too (so you could theoretically generate automatically the A100, A200, A400 and A700 of every custom color you want).

**Comment by Dinuz** [comment]
Note that the css variable will require to use --v-theme prefix, otherwise are not recognized anymore.

**Comment by ricardovanlaarhoven** [comment]
I was trying to update the docs for this. But while doing this I was wondering if I myself had issues with this.
And i remembered writing this code with no issues.

So i'm wondering what exactly the issue is. I followed the docs and created this

```
const customTheme: ThemeDefinition = {
  dark: true,
  colors: {
    background: '#1c2123',
    primary: '#f59740',
  },
}

export default createVuetify({
  theme: {
    defaultTheme: 'customTheme',
    themes: {
      customTheme,
    }
  },
  ```

Even when adding
```

    variations: {
      colors: ['primary'],
      lighten: 1,
      darken: 2,
    },
```

evrything works fine
![image](https://user-images.githubusercontent.com/1246691/195020885-aff8163f-7040-4ce4-83ef-337a42476465.png)

**Comment by c-harding** [comment]
@ricardovanlaarhoven The issue is for colours that do not appear in the default themes. Try renaming the color `primary` to `custom`, and adding that to your variations

**Comment by ricardovanlaarhoven** [comment]
If that's the case this issue shouldn't be labelled as `documentation`? @KaelWD 

Or does vuetify want that you only use color variations with the default colors or the default themes?

**Comment by c-harding** [comment]
More specifically, the problem is that variations apply to all installed themes, and the themes `light` and `dark` are automatically installed unless overridden. The variation colors need to exist in all themes.

The easiest https://github.com/vuetifyjs/vuetify/labels/T%3A%20documentation solution is to add a note to the docs that if custom colors are used for variations, then the themes `light` and `dark` also need to be overridden.

However, I would also like to see a clearer error message: it’s currently `Colors can only be numbers or strings, recieved undefined instead`, but something more specific such as `Variation colors must be defined in all themes, color "custom" is undefined in theme "light"` would be better. Even better would be to provide an option to disable the default themes, and to mention this option in the error message.

**Comment by johnleider** [comment]
fixed by #15872
